modules/aws-iam-policies/main.tf

```diff
@@ -23,10 +24,19 @@ data "aws_iam_policy_document" "emr_creator_policy" {
       "elasticmapreduce:TerminateJobFlows",
       "iam:PassRole"
     ]
-    resources = [
-      "arn:${var.arn_partition}:iam::${data.aws_caller_identity.current.account_id}:role/*",
-      "arn:${var.arn_partition}:elasticmapreduce:*:${data.aws_caller_identity.current.account_id}:cluster/*"
-    ]
+    resources = flatten([
+                  length(var.tamr_emr_role_arns)==0 ?
```

Comment:
Unfortunately I think you're right that these ternary operators are necessary to pull in those data objects for the default behavior.